pinot-core/src/main/java/org/apache/pinot/core/operator/InstanceResponseOperator.java

```diff
@@ -45,8 +46,39 @@ protected InstanceResponseBlock getNextBlock() {
     DataTable dataTable = instanceResponseBlock.getInstanceResponseDataTable();
 
     mainThreadTimer.stop();
-    long totalThreadCpuTimeNs =
-        intermediateResultsBlock.getExecutionThreadCpuTimeNs() + mainThreadTimer.getThreadTimeNs();
+    long endWallClockTimeNs = System.nanoTime();
+
+    long singleThreadCpuTimeNs = mainThreadTimer.getThreadTimeNs();
+    long multipleThreadCpuTimeNs = intermediateResultsBlock.getExecutionThreadCpuTimeNs();
```

Comment:
Just a note that if this is backed by `ThreadMXBean.getCurrentThreadCpuTime` and the JVM is not running on bare metal, this call can take up to half a millisecond.

Reply:
@richardstartin  we have a config to turn the thread cpu measurements off. If there is a way of auto-detecting this case and turning it off after the server is started, we can add the code. Otherwise, we can turn it off before starting the server.